app/javascript/src/Stats/inlinechart/index.jsx

```diff
@@ -0,0 +1,137 @@
+// @flow
+
+type Props = {
+    endPoint: string,
+    metricName: string,
+    title: string,
+    unitPluralized: string
+}
+
+type State = {
+    loading: boolean,
+    title: string,
+    total: string,
+    values: Array<number>,
+    unit: string,
+    unitPluralized: string
+}
+
+type RefObject = { current: null | HTMLDivElement }
+
+type DataMetric = {
+    unit: string
+}
+
+type Data = {
+    metric: DataMetric,
+    total: number,
+    values: Array<number>
+}
+
+type ResponseError = {
+  status: string,
+  statusText: string
+}
+
+import React, {Component} from 'react'
+import c3 from 'c3'
+import 'whatwg-fetch'
+import 'core-js/es6/promise'
+import 'url-polyfill'
+import moment from 'moment'
+
+class InlineChart extends Component<Props, State> {
+  c3ChartContainer: RefObject
+
+  constructor (props: Props) {
+    super(props)
+    this.c3ChartContainer = React.createRef()
+    this.state = {
+      loading: true,
+      title: this.props.title,
+      total: '',
+      values: [],
+      unit: '',
+      unitPluralized: this.props.unitPluralized
+    }
+  }
+
+  generateC3Chart () {
```

Comment:
Have you look for a c3 flow type? I'm not sure if it exist within the lib or any other flow type lib, but might be worth it a quick look

Reply:
Couldn't find anything like a "C3 type" :(